tests/integration/test_metrics.py

```diff
@@ -1880,3 +1879,121 @@ def test_metric_bucket_encoding_dynamic_global_config_option(
     else:
         assert metrics[dname]["value"] == [1337.0]
         assert metrics[sname]["value"] == [42.0]
+
+
+def test_relay_forwards_events_without_extracting_metrics_on_broken_global_filters(
+    mini_sentry,
+    relay_with_processing,
+    transactions_consumer,
+    metrics_consumer,
+):
+    metrics_consumer = metrics_consumer()
+    tx_consumer = transactions_consumer()
+
+    mini_sentry.global_config["filters"] = {
+        "version": "Halp! I'm broken!",
+        "filters": [],
+    }
+
+    project_id = 42
+    mini_sentry.add_full_project_config(project_id)
+    config = mini_sentry.project_configs[project_id]["config"]
+    config["transactionMetrics"] = {
+        "version": 1,
+    }
+
+    relay = relay_with_processing(
+        options={
+            "aggregator": {
+                "bucket_interval": 1,
+                "initial_delay": 0,
+                "debounce_delay": 0,
+                "shift_key": "none",
+            }
+        }
+    )
+    transaction = generate_transaction_item()
+    relay.send_transaction(project_id, transaction)
+
+    tx, _ = tx_consumer.get_event()
+    assert tx is not None
```

Comment:
`get_event` already asserts the transaction is `None`, but that's not super clear so I added the check here for readability.